File: src/test/java/no/nav/sosialhjelp/soknad/business/service/minesaker/MineSakerMetadataServiceTest.java
package no.nav.sosialhjelp.soknad.business.service.minesaker;

import no.nav.sosialhjelp.soknad.business.db.repositories.soknadmetadata.SoknadMetadataRepository;
import no.nav.sosialhjelp.soknad.business.domain.SoknadMetadata;
import no.nav.sosialhjelp.soknad.domain.model.kravdialoginformasjon.SoknadType;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;

import java.time.LocalDateTime;
import java.util.Collections;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.when;

@ExtendWith(MockitoExtension.class)
class MineSakerMetadataServiceTest {

    @Mock
    private SoknadMetadataRepository soknadMetadataRepository;

    @InjectMocks
    private MineSakerMetadataService mineSakerMetadataService;

    @Test
    void skalHenteInnsendteSoknaderForBruker() {
        var soknadMetadata = new SoknadMetadata();
        soknadMetadata.fnr = "12345";
        soknadMetadata.behandlingsId = "beh123";
        soknadMetadata.type = SoknadType.SEND_SOKNAD_KOMMUNAL;
        soknadMetadata.innsendtDato = LocalDateTime.now();

        when(soknadMetadataRepository.hentAlleInnsendteSoknaderForBruker("12345"))
                .thenReturn(Collections.singletonList(soknadMetadata));

        var dtos = mineSakerMetadataService.hentInnsendteSoknader("12345");

        assertThat(dtos).hasSize(1);
    }

    @Test
    void skalReturnereTomListeVedNull() {
        when(soknadMetadataRepository.hentAlleInnsendteSoknaderForBruker("12345"))
                .thenReturn(null);

        var dtos = mineSakerMetadataService.hentInnsendteSoknader("12345");

        assertThat(dtos).isEmpty();
    }

    @Test
    void skalReturnereTomListeVedTomListe() {
        when(soknadMetadataRepository.hentAlleInnsendteSoknaderForBruker("12345"))
                .thenReturn(Collections.emptyList());

        var dtos = mineSakerMetadataService.hentInnsendteSoknader("12345");

        assertThat(dtos).isEmpty();
    }
}

File: src/test/kotlin/no/nav/sosialhjelp/soknad/api/minesaker/MineSakerMetadataServiceTest.kt
package no.nav.sosialhjelp.soknad.api.minesaker

import io.mockk.every
import io.mockk.mockk
import no.nav.sosialhjelp.soknad.db.repositories.soknadmetadata.SoknadMetadata
import no.nav.sosialhjelp.soknad.db.repositories.soknadmetadata.SoknadMetadataRepository
import no.nav.sosialhjelp.soknad.db.repositories.soknadmetadata.SoknadMetadataType
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import java.time.LocalDateTime

internal class MineSakerMetadataServiceTest {
    private val soknadMetadataRepository: SoknadMetadataRepository = mockk()
    private val mineSakerMetadataService = MineSakerMetadataService(soknadMetadataRepository)

    @Test
    fun skalHenteInnsendteSoknaderForBruker() {
        val soknadMetadata = SoknadMetadata(
            id = 0L,
            behandlingsId = "beh123",
            fnr = "12345",
            type = SoknadMetadataType.SEND_SOKNAD_KOMMUNAL,
            opprettetDato = LocalDateTime.now(),
            sistEndretDato = LocalDateTime.now(),
            innsendtDato = LocalDateTime.now(),
        )

        every { soknadMetadataRepository.hentAlleInnsendteSoknaderForBruker("12345") } returns listOf(soknadMetadata)

        val dtos = mineSakerMetadataService.hentInnsendteSoknader("12345")
        assertThat(dtos).hasSize(1)
    }

    @Test
    fun skalReturnereTomListeVedTomListe() {
        every { soknadMetadataRepository.hentAlleInnsendteSoknaderForBruker("12345") } returns emptyList()

        val dtos = mineSakerMetadataService.hentInnsendteSoknader("12345")
        assertThat(dtos).isEmpty()
    }
}
